src/Apache.IoTDB/SessionPool.cs

```diff
@@ -101,12 +175,86 @@ public async Task Open(CancellationToken cancellationToken = default)
         {
             _clients = new ConcurrentClientQueue();
             _clients.Timeout = _timeout * 5;
-            for (var index = 0; index < _poolSize; index++)
+            if (_nodeUrls.Count == 0)
+            {
+                for (var index = 0; index < _poolSize; index++)
+                {
+                    _clients.Add(await CreateAndOpen(_host, _port, _enableRpcCompression, _timeout, cancellationToken));
+                }
+            }
+            else
+            {
+                foreach (var endPoint in _endPoints)
+                {
+                    try
+                    {
+                        for (var index = 0; index < _poolSize; index++)
+                        {
+                            var client = await CreateAndOpen(endPoint.Ip, endPoint.Port, _enableRpcCompression, _timeout, cancellationToken);
+                            _clients.Add(client);
+                        }
+                        break;
+                    }
+                    catch (Exception e) // CreateAndOpen will not throw TException
+                    {
+#if NET461_OR_GREATER || NETSTANDARD2_0
+#else
+                        _clients.ClientQueue.Clear();
+#endif
+                        continue;
+                    }
+                }
+                if (_clients.ClientQueue.Count != _poolSize)
+                {
+                    throw new TException("Error occurs when opening session pool. Could not connect to any server", null);
+                }
+            }
+        }
+        public async Task Reconnect(CancellationToken cancellationToken = default)
+        {
+            if (_nodeUrls.Count == 0)
+            {
+                await Open(_enableRpcCompression);
+                return;
+            }
+
+            bool isConnected = false;
+            Random random = new Random();
+
+            for (int i = 0; i < RetryNum && !isConnected; i++)
+            {
+                int currentHostIndex = random.Next(0, _endPoints.Count);
+                int attempts = 0;
+
+                while (attempts < _endPoints.Count)
+                {
+                    int j = (currentHostIndex + attempts) % _endPoints.Count;
+
+                    try
+                    {
+                        for (int index = 0; index < _poolSize; index++)
```

Comment:
Why do we need to reconnect all the pools？
Different clients can connect different nodes.
Instead of connecting clients in the entire session pool to the same endpoint